xds/src/main/java/io/grpc/xds/LookasideChannelLb.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.logging.Level.FINEST;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment.Policy.DropOverload;
+import io.envoyproxy.envoy.api.v2.endpoint.LocalityLbEndpoints;
+import io.envoyproxy.envoy.type.FractionalPercent;
+import io.envoyproxy.envoy.type.FractionalPercent.DenominatorType;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import io.grpc.internal.ExponentialBackoffPolicy.Provider;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.xds.LoadReportClient.LoadReportCallback;
+import io.grpc.xds.LoadReportClientImpl.LoadReportClientFactory;
+import io.grpc.xds.LocalityStore.LocalityStoreImpl;
+import io.grpc.xds.XdsComms.AdsStreamCallback;
+import io.grpc.xds.XdsComms.LbEndpoint;
+import io.grpc.xds.XdsComms.LocalityInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * A load balancer that has a lookaside channel. This layer of load balancer creates a channel to
+ * the remote load balancer. LrsClient, LocalityStore and XdsComms are three branches below this
+ * layer, and their implementations are provided by their factories.
+ */
+final class LookasideChannelLb extends LoadBalancer {
+
+  private final ManagedChannel lbChannel;
+  private final LoadReportClient lrsClient;
+  private final XdsComms2 xdsComms2;
+
+  @VisibleForTesting
+  LookasideChannelLb(
+      Helper helper,
+      AdsStreamCallback adsCallback,
+      String balancerName,
+      LoadReportClientFactory lrsClientFactory,
+      LoadBalancerRegistry lbRegistry,
+      LocalityStoreFactory localityStoreFactory) {
+    lbChannel = initLbChannel(helper, balancerName);
+    final LocalityStore localityStore = localityStoreFactory.newLocalityStore(helper, lbRegistry);
```

Comment:
Given the current structure, we probably no longer need factories for `LocalityStore` and `LoadReportClient`. They can be directly injected in the secondary constructor.

Reply:
Done.